### PR TITLE
Share dialog under13

### PR DESCRIPTION
**Pull request by Bjvanminnen**
This is a fair amount of code change, with just a small amount of functional change. Focus on the last commit to just get the functional change.

When viewing a shared app, there are two things we potentially want to prompt you about.
(1) If we don't know how old you are we want to prompt you for your age.
(2) If the app uses data APIs we want to warn you about this once per app.

When this work was initially done, if you were logged in and < 13 we would do a redirect if you tried to go to anything applab related (including a shared app). That is no longer the case if you have a teacher that has accepted the TOS. This resulted in us prompting you for your age even though we already know it (and don't actually want to block you if it's < 13).

New logic is as follows:
(1) If the App has no Data APIs, we never need to prompt you for your age or warn you about Data
(2) If the App has Data APIs, and you're not logged in, we'll prompt you for your age (and block you if you say < 13).
(3) If the App has Data APIs, and you're logged in, we don't prompt you for your age, depending on dashboard to have redirected you if you <13 and you don't have a teacher who has accepted the TOS.

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
I've run this branch against circle quite a few times at this point. This most recent one only had one failure (applab_templateBacked - failed because of a tab crash). An earlier run only had applab_scenarios fail, so I think this can be considered green.

File: apps/test/unit/shareWarningsDialogTest.js [hunk 1]
```diff
@@ -98,10 +98,10 @@ describe('ShareWarningsDialog', function () {
     assert.strictEqual(ageDropdowns.length, 1);
   });
 
-  it('doesnt show a dialog when is13Plus and app doesnt store data', function () {
+  it('doesnt show a dialog when note promptForAge and app doesnt store data', function () {
```
Comment (caleybrock) [inline apps/test/unit/shareWarningsDialogTest.js]: spelling: note -> not

**Comment by caleybrock** [comment]
lgtm

**Comment by Bjvanminnen** [comment]
Closing to reopen against staging-next

**Pull request closed without merging**